src/Microsoft.ComponentDetection.Detectors/go/GoComponentDetector.cs

```diff
@@ -78,7 +73,7 @@ public class GoComponentDetector : FileComponentDetector
             var goModFileContents = reader.ReadToEnd();
             goModFile.Stream.Dispose();
 
-            return this.CheckGoModVersion(goModFileContents, processRequest, goModFile);
+            return GoModFileParser.ShouldParseGoSumFile(goModFileContents, processRequest, goModFile, this.Logger);
```

Comment:
it seems weird that having a `GoSumFileParser` is the `GoModFileParser` that one that tells you if you have to parse the `go.sum` file or not. There are different ways that you can handle that one is that you have a method in the detector class that is the one that knows when and how the different parser should interact. Then the `GoModFileParser` will have a method to output the version. The coordinator method then is the one that contains the business logic to determine if the `go.sum` should be parsed or not. In that way you keep the `GoModFileParser` concise.